File: android/RecyclerView-master/AgriinformaticsMAD/app/src/main/java/com/dicoding/agriinformaticsmad/MainActivity.java
package com.dicoding.agriinformaticsmad;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private EditText edtName;
    private EditText edtAge;
    private EditText edtGender;
    private EditText edtAddress;
    private Button edtButton;
    private Button btnActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // binding onto variables
        edtName = findViewById(R.id.edtNama);
        edtAge = findViewById(R.id.edtUmur);
        edtGender = findViewById(R.id.edtGender);
        edtAddress = findViewById(R.id.edtAlamat);
        edtButton = findViewById(R.id.btnEnter);

        // add onClick function to enable button
        edtButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        // Use toast to show Name as a Toast notification
        Toast.makeText(MainActivity.this, edtName.getText(), Toast.LENGTH_SHORT).show();

        // sending an intent to open Receiving Activity
        Intent intent = new Intent(MainActivity.this, ReceivingActivity.class);

        // sending data as an intent to Receiving Activity
        intent.putExtra("Name", edtName.getText().toString());
        intent.putExtra("Age", edtAge.getText().toString());
        intent.putExtra("Gender", edtGender.getText().toString());
        intent.putExtra("Address", edtAddress.getText().toString());

        // start intent
        startActivity(intent);

    }
}

File: android/MAD2/app/src/main/java/com/frhnfath/mad2/MainActivity.kt
package com.frhnfath.mad2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var edtName: EditText
    private lateinit var edtAge: EditText
    private lateinit var edtGender: EditText
    private lateinit var edtAddress: EditText
    private lateinit var buttonEnter: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        edtName = findViewById(R.id.edtTextNama)
        edtAge = findViewById(R.id.edtTextUmur)
        edtGender = findViewById(R.id.edtTextGender)
        edtAddress = findViewById(R.id.edtTextAlamat)
        buttonEnter = findViewById(R.id.btnEnter)

        buttonEnter.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btnEnter -> {
                val intent = Intent(this@MainActivity, DetailActivity::class.java)
                startActivity(intent)
            }
        }
    }
}